src/xtgeo/xyz/_xyz.py

```diff
@@ -179,256 +152,193 @@ def from_file(self, pfile, fformat="xyz"):
-        if fformat in ["xyz", "poi", "pol"]:
-            _xyz_io.import_xyz(self, pfile.name)
-        elif fformat == "zmap":
-            _xyz_io.import_zmap(self, pfile.name)
-        elif fformat in ("rms_attr", "rmsattr"):
-            _xyz_io.import_rms_attr(self, pfile.name)
-        else:
-            logger.error("Invalid file format (not supported): %s", fformat)
-            raise ValueError("Invalid file format (not supported): %s", fformat)
-
-        logger.info("Reading from file %s... done", pfile.name)
-        logger.debug("Dataframe head:\n%s", self._df.head())
-        self._filesrc = pfile.name
-
-        return self
-
+    @abstractmethod
     def from_list(self, plist):
-        """Import Points or Polygons from a list.
+        """Create Points or Polygons from a list-like input (deprecated).
+
+        The following inputs are possible:
 
-        [(x1, y1, z1, <id1>), (x2, y2, z2, <id2>), ...]
+        * List of tuples [(x1, y1, z1, <id1>), (x2, y2, z2, <id2>), ...].
+        * List of lists  [[x1, y1, z1, <id1>], [x2, y2, z2, <id2>], ...].
+        * List of numpy arrays  [nparr1, nparr2, ...] where nparr1 is first row.
+        * A numpy array with shape [??1, ??2] ...
+        * An existing pandas dataframe
 
         It is currently not much error checking that lists/tuples are consistent, e.g.
         if there always is either 3 or 4 elements per tuple, or that 4 number is
         an integer.
 
         Args:
-            plist (str): List of tuples, each tuple is length 3 or 4
+            plist (str): List of tuples, each tuple is length 3 or 4.
 
         Raises:
             ValueError: If something is wrong with input
 
-        .. versionadded: 2.6
+        .. versionadded:: 2.6
+        .. versionchanged:: 2.16
+        .. deprecated:: 2.16
+           Use e.g. xtgeo.Points(list_like).
         """
+        ...
 
-        first = plist[0]
-        if len(first) == 3:
-            self._df = pd.DataFrame(
-                plist, columns=[self._xname, self._yname, self._zname]
-            )
+    def delete_columns(self, clist, strict=False):
+        """Delete one or more columns by name in a safe way for Points or Polygons.
 
-        elif len(first) == 4:
-            self._df = pd.DataFrame(
-                plist, columns=[self._xname, self._yname, self._zname, self._pname]
-            )
-        else:
-            raise ValueError(
-                "Wrong length detected of first tuple: {}".format(len(first))
-            )
-        self._df.dropna(inplace=True)
-
-    def to_file(
-        self,
-        pfile,
-        fformat="xyz",
-        attributes=False,
-        pfilter=None,
-        wcolumn=None,
-        hcolumn=None,
-        mdcolumn="M_MDEPTH",
-        **kwargs,
-    ):  # pylint: disable=redefined-builtin
-        """Export XYZ (Points/Polygons) to file.
+        Note that the coordinate columns will be protected, as well as then
+        POLY_ID column (pname atribute) if Polygons.
```

Comment:
I think this must be an `@abstractmethod` as it here (see docstring) operates with different protected columns if Points or Polygons. Hence it is not "agnostic".

Reply:
So the way this was handled was that `Polygons` object overrode the `delete_columns`. However, I guess it was perhaps not clear what was going on and it technically overrode it in a way that was not compatible with `XYZ`. Changed it so that the overriding now is opaque to whether you have a `Points`, `Polygons` or any other `XYZ`.